src/analysis/history_utils.hpp

```diff
@@ -154,6 +154,24 @@ KOKKOS_INLINE_FUNCTION Real CalcEMMomentumFlux(Pack &pack, Geometry &geom,
   return -T_EM_11 * lapse * gdet;
 } // momentum flux
 
+template <typename Pack, typename Geometry>
+KOKKOS_INLINE_FUNCTION Real CalcMagneticFluxPhi(Pack &pack, Geometry &geom,
+                                                const int pvel_lo, const int pvel_hi,
+                                                const int pb_lo, const int pb_hi,
+                                                const int b, const int k, const int j,
+                                                const int i) {
+  Real gam[3][3];
+  geom.Metric(CellLocation::Cent, k, j, i, gam);
+  Real gdet = geom.DetGamma(CellLocation::Cent, k, j, i);
+  Real lapse = geom.Lapse(CellLocation::Cent, k, j, i);
+
+  const Real Bp[] = {pack(b, pb_lo, k, j, i), pack(b, pb_lo + 1, k, j, i),
+                     pack(b, pb_hi, k, j, i)};
+
+  // \int B^r sqrt(-gdet) := B^r / alpha * (alpha * detgamma)
+  return std::abs(Bp[0]) * gdet;
```

Comment:
The primitive `B` field in Phoebus is densitized---it's multiplied by a factor of `detgam`. So you don't actually need to multiply here by `gdet`. I think that's what's wrong. 

I *think* that that implies this whole function could just be replaced with `return std::abs(pack(b, pb_lo, k, j, i));`

Reply:
Ahhh I see. So is the primitive `detgam B / alpha`, regarding the conversations above?

Reply:
Ah... I think I've confused myself. Looking again at the phoebus doc:
- Primitive variable is just $B^i$
- Conserved variable is $\sqrt{\gamma} B^i$

and this is the difference @brryan was referring to but he got it backwards. In HARM the conserved variable is 

I'm now looking at `prim2con` and the phoebus doc. The HARM conserved variable is $\sqrt{g} B^i$, but as far as I can tell the primitive variables are identical. I think this is what @brryan was referring to before.

So if you use the `fluid_cons` instead of `fluid_prim` then all you need to do is multiply by the lapse to get $\sqrt{g}B^i$ and then you're done. I.e.,
```C++
// assumes that pack is packing conserved B not primitive B
// conserved B is \sqrt{\gamma} B^k
// we want \sqrt{g} B^k = \sqrt{gamma} alpha B^k
return std::abs(pack(b, pb_lo, k, j, i))*lapse;
```

Sorry for the confusion! I think me and @brryan spouted half truths that made sense in our head without sitting down and writing it out

Reply:
Also... nice latex works now in github. When did that happen?

Reply:
Ah perfect! Will implement these tomorrow

Reply:
Edited to take all the blame instead of @brryan sorry Ben. ;)